packages/client/src/rtc/Call.ts

```diff
@@ -194,6 +194,7 @@ export class Call {
   }
 
   private readonly streamClientBasePath: string;
+  private streamClientEventHandlers = new Map<Function, CallEventHandler>();
```

Comment:
Hm, I'd think that technically it's possible to leave and rejoin a call, no? I'd expect that registered event handlers would survive the rejoin.

Reply:
Ah, ok, I see that we remove all other event handlers in leave

Reply:
I'm thinking about moving the "event registration" for some of the events out from the Constructor. Ideally, these registrations shall be done either in `join()` or `watch()`.

Reply:
I've looked into it, and I think it's incorrect to remove coordinator event handlers in `leave`; those event handlers don't get invalid just because we leave the call. I think if an integrator registers an event with `call.on` they should be responsible for calling `call.off` when they don't need those handlers anymore. If an integrator supports a use-case where you can transition between the following call states: watching -> joined -> watching it'd be unexpected that they would have to reregister their event handlers when they leave the call.

Reply:
I see. In that case, let's keep everything as you propose